src/specBuilder/bullet/bulletSpecBuilder.ts

```diff
@@ -113,19 +132,129 @@ export function getBulletMarks(props: BulletSpecProps): Mark[] {
+            "field": "index",
+            "mult": 60,
+            "offset": -53
+          },
+          "y2": {
+            "field": "index",
+            "mult": 60,
+            "offset": -29
+          },
           "stroke": {"value": `${solidColor}`},
-          "strokeWidth": {"value": 2},
+          "strokeWidth": {"value": 2}
         }
       }
     }
   ]
+
+  const horizontalMarks : GroupMark[] = [
+    {
+      "name": "bulletGroup",
+      "type": "group",
+      "from": {
+        "facet": {
+          "data": "table",
+          "name": "bulletGroups",
+          "groupby": `${props.dimension}`
+        }
+      },
+      "signals": [{ "name": "width", "update": "bandwidth('xGroupScale')" }],
+      "encode": {
+        "update": {
+          "x": { "scale": "xGroupScale", "field": `${props.dimension}` },
+          "height": {"value": 20},
+          "width": { "signal": "bandwidth('xGroupScale')" }
+        }
+      },
+      "scales": [
+        {
+          "name": "xscale",
+          "type": "linear",
+          "domain": [0, { "signal": "data('max_values')[0].maxOverall" }],
+          "range": [0, { "signal": "width" }],
+          "round": true,
+          "zero": true
+        }
+      ],
+      "marks": [
+        {
+          "name": "bullet0rect",
+          "type": "rect",
+          "from": { "data": "bulletGroups" },
+          "encode": {
+            "enter": {
+              "x": { "scale": "xscale", "value": 0 },
+              "x2": { "scale": "xscale", "field": `${props.metric}` },
+              "y": { "value": 21 },
+              "height": {"value": 6},
+              "fill": { "value": `${props.color}` },
+              "cornerRadiusTopRight": { "value": 2 },
+              "cornerRadiusBottomRight": { "value": 2 }
+            }
+          }
+        },
+        {
+          "type": "text",
+          "name": "bullet0barlabel",
+          "from": { "data": "bulletGroups" },
+          "description": "graphLabel",
+          "encode": {
+            "enter": {
+              "x": { "value": 0 },
+              "y": {"value": 5},
+              "text": {"field": `${props.dimension}`},
+              "align": {"value": "left"},
+              "baseline": {"value": "bottom"},
+              "fill": {"value": `${barLabelColor}`},
+              "fontSize": {"value": 11.5}
+            }
+          }
+        },
+        {
+          "type": "text",
+          "from": { "data": "bulletGroups" },
+          "name": "bullet0amountlabel",
+          "encode": {
+            "enter": {
+              "x": { "signal": "width", "offset": 0 },
+              "y": { "value": 5 }, 
+              "text": { "field": `${props.metric}` },
+              "align": { "value": "right" }, 
+              "baseline": { "value": "bottom" },
+              "fontSize": { "value": 11.5 },
+              "fill": { "value": `${solidColor}` }
+            }
+          }
+        },
+        {
+          "type": "rule",
```

Comment:
Names like this should be dynamic and follow our mark naming convention. `bullet0_target` which is actually `` `${props.name}_target` ``

Reply:
Sorry this comment should be for line 242